src/Hid.Shared/Hid.Helpers.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.ComponentModel;
+    using System.Runtime.InteropServices;
+    using System.Text;
+    using Microsoft.Win32.SafeHandles;
+
+    /// <content>
+    /// Methods and nested types that are not strictly P/Invokes but provide
+    /// a slightly higher level of functionality to ease calling into native code.
+    /// </content>
+    public static partial class Hid
+    {
+        public static Guid HidD_GetHidGuid()
+        {
+            Guid guid;
+            HidD_GetHidGuid(out guid);
+            return guid;
+        }
+
+        public static HiddAttributes HidD_GetAttributes(SafeFileHandle hFile)
+        {
+            var result = HiddAttributes.Create();
+            if (!HidD_GetAttributes(hFile, ref result))
+            {
+                throw new Win32Exception();
+            }
+
+            return result;
+        }
+
+        public static SafePreparsedDataHandle HidD_GetPreparsedData(SafeFileHandle hDevice)
+        {
+            SafePreparsedDataHandle preparsedDataHandle;
+            if (!HidD_GetPreparsedData(hDevice, out preparsedDataHandle))
+            {
+                throw new Win32Exception();
+            }
+
+            return preparsedDataHandle;
+        }
+
+        public static HidpCaps HidP_GetCaps(SafePreparsedDataHandle preparsedData)
+        {
+            var hidCaps = default(HidpCaps);
+            var result = HidP_GetCaps(preparsedData, ref hidCaps);
+            switch (result)
+            {
+                case NTStatus.HIDP_STATUS_SUCCESS:
+                    return hidCaps;
+
+                case NTStatus.HIDP_STATUS_INVALID_PREPARSED_DATA:
+                    throw new ArgumentException("The specified preparsed data is invalid.", nameof(preparsedData));
+
+                default:
+                    result.ThrowOnError();
+                    return default(HidpCaps);
```

Comment:
In the event that ThrowOnError doesn't throw (presumably because it was not in fact an error), do you really want to return an uninitialized struct, or would returning `hidCaps` be more appropriate?